File: ruby/json_object_with_array_values.rb
# Script takes arguments and transforms them into dict with arrays as dict values
# and returns as JSON
require 'json'

my_strings = ARGV

# Make dict with the string as key and list of letters as value
string_letters_dict = Hash[my_strings.map { |string| [string, string.chars.map(&:upcase)]}]

# Cast to JSON and print to stdout
puts JSON.generate(string_letters_dict)

File: php/json_object_with_array_values.php
<?php
// Script takes args and transforms into python dict with arrays as dict values

// Get the command-line arguments into an array
$myStrings = array_slice($argv, 1);

// Create an associative array (dictionary) with each string as a key and an array of letters as the value
$stringLettersDict = array_combine($myStrings, array_map('str_split', $myStrings));

// Convert each letter to uppercase within the arrays
foreach ($stringLettersDict as &$lettersArray) {
    $lettersArray = array_map('strtoupper', $lettersArray);
}

// Encode the dictionary as JSON and print to stdout
echo json_encode($stringLettersDict) . PHP_EOL;
?>